apps/costcontrol/js/message_handler.js

```diff
@@ -323,26 +272,74 @@
     });
   }
 
+  function getCallback(activity, app) {
+    var activityName, action;
+    switch (activity) {
+      case 'lowBalance':
+      case 'zeroBalance':
+        activityName = 'costcontrol/balance';
+        action = function() {
+          if (window.parent.BalanceTab) {
+            window.parent.CostControlApp.showBalanceTab();
+          }
+        };
+        break;
+      case 'topUpError' :
+        activityName = 'costcontrol/balance';
+        action = function() {
+          if (window.parent.BalanceTab) {
+            window.parent.BalanceTab.topUpWithCode(true);
+          }
+        };
+        break;
+      default:
+        activityName = 'costcontrol/data_usage';
+        action = function() {
+          if (window.parent.CostControlApp) {
+            window.parent.CostControlApp.showDataUsageTab();
+          }
+        };
+        break;
+    }
+
+    if (!inStandAloneMode()) {
+      return function() {
+        if (inApplicationMode()) {
+          app.launch();
+          action && action();
+        } else {
+          activity = new MozActivity({name: activityName});
+        }
+      };
+    }
+    return;
```

Comment:
To keep consistency, return a no-op function instead of `undefined`. Define a `noop()` function and return it.